src/hardware/delay.rs

```diff
@@ -0,0 +1,34 @@
+use super::hal::hal::blocking::delay::{DelayMs, DelayUs};
+
+pub struct AsmDelay {
+    frequency_us: u32,
+    frequency_ms: u32,
+}
+
+impl AsmDelay {
+    pub fn new(freq: u32) -> AsmDelay {
+        AsmDelay {
+            // multiply by two to get to the frequency that halt instructions have to be issued on the STM32H7 (dual-issue pipeline)
```

Comment:
?

I think you are often confusing yourself with too many inaccurate, badly named, half-done things.
What's `frequency_us` and `_ms`? What do they mean, what are the units? Why would you need both? What happens if a multiplication overflows? And when? Why would you still need workarounds like multiplications by two if `cortex_m::asm::delay()` already guarantees a lower bound?

Reply:
I was just adapting this from booster which itself is basically the same as `asm_delay` except for the missing division by two. I agree that the name does not make much sense and I should not have tried to stick with it. As I was stating in the issue, using `cortex_m::asm::delay()` and waiting for the specified cycles leads to exactly half the delay time. 
Would you like me to adapt this implementation with sensible names? Or could you point me to a fitting crate for this?

Reply:
Ah. I misunderstood you then. If `corex_m::asm::delay(t)` delays by less then `t` cycles (`~t/2` as you say) then that's a bug in `cortex_m` as it breaks its promise. File an issue?

Reply:
Issue filed. See code.